### ISSUE TITLE
the effect is not very good

### ISSUE THREAD

**Issue by Ruinmou** [skycrapers/TecoGAN-PyTorch#26]
run `bash ./test.sh BD TecoGAN/TecoGAN_VimeoTecoGAN_4xSR_2GPU`
results
![13880](https://user-images.githubusercontent.com/83195382/148916491-129ef175-9b73-41fc-96b1-d216b77a80f0.png)

**Comment by jndeng** [comment]
Hi @hai8023,

Which dataset are you using? I've tested this model on Vid4 and ToS3, and I got decent results.

**Comment by Ruinmou** [comment]
Hello, both ToS3 Vid4 are not very good @jndeng

**Comment by jndeng** [comment]
That's weird. Which pretrained weight are you using? For ToS3 and Vid4 dataset, it should be [BD-4x-Vimeo](https://drive.google.com/file/d/13FPxKE6q7tuRrfhTE7GB040jBeURBj58/view?usp=sharing).

**Comment by Ruinmou** [comment]
TecoGAN_4x_BD_Vimeo_iter500K.pth
I'm using the right one.
`/home/haimou/anaconda3/envs/TecoGAN-T/lib/python3.8/site-packages/torch/cuda/__init__.py:125: UserWarning: 
NVIDIA GeForce RTX 3090 with CUDA capability sm_86 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_61 sm_70 sm_75 compute_37.
If you want to use the NVIDIA GeForce RTX 3090 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

  warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))
2022-01-12 09:27:42,454 [INFO]: scale: 4
2022-01-12 09:27:42,454 [INFO]: manual_seed: 0
2022-01-12 09:27:42,454 [INFO]: verbose: False
2022-01-12 09:27:42,454 [INFO]: dataset:
2022-01-12 09:27:42,454 [INFO]:   degradation:
2022-01-12 09:27:42,455 [INFO]:     type: BD
2022-01-12 09:27:42,455 [INFO]:     sigma: 1.5
2022-01-12 09:27:42,455 [INFO]:   test1:
2022-01-12 09:27:42,455 [INFO]:     name: Vid4
2022-01-12 09:27:42,455 [INFO]:     gt_seq_dir: data/Vid4/GT
2022-01-12 09:27:42,455 [INFO]:     lr_seq_dir: data/Vid4/Gaussian4xLR
2022-01-12 09:27:42,455 [INFO]:     num_worker_per_gpu: 3
2022-01-12 09:27:42,455 [INFO]:     pin_memory: True
2022-01-12 09:27:42,455 [INFO]:   test2:
2022-01-12 09:27:42,455 [INFO]:     name: ToS3
2022-01-12 09:27:42,455 [INFO]:     gt_seq_dir: data/ToS3/GT
2022-01-12 09:27:42,455 [INFO]:     lr_seq_dir: data/ToS3/Gaussian4xLR
2022-01-12 09:27:42,455 [INFO]:     num_worker_per_gpu: 3
2022-01-12 09:27:42,455 [INFO]:     pin_memory: True
2022-01-12 09:27:42,455 [INFO]: model:
2022-01-12 09:27:42,455 [INFO]:   name: TecoGAN
2022-01-12 09:27:42,455 [INFO]:   generator:
2022-01-12 09:27:42,456 [INFO]:     name: FRNet
2022-01-12 09:27:42,456 [INFO]:     in_nc: 3
2022-01-12 09:27:42,456 [INFO]:     out_nc: 3
2022-01-12 09:27:42,456 [INFO]:     nf: 64
2022-01-12 09:27:42,456 [INFO]:     nb: 10
2022-01-12 09:27:42,456 [INFO]:     load_path: pretrained_models/TecoGAN_4x_BD_Vimeo_iter500K.pth
2022-01-12 09:27:42,456 [INFO]:     load_path_lst: ['pretrained_models/TecoGAN_4x_BD_Vimeo_iter500K.pth']
2022-01-12 09:27:42,456 [INFO]: test:
2022-01-12 09:27:42,456 [INFO]:   save_res: True
2022-01-12 09:27:42,456 [INFO]:   res_dir: results
2022-01-12 09:27:42,456 [INFO]:   padding_mode: reflect
2022-01-12 09:27:42,456 [INFO]:   num_pad_front: 5
2022-01-12 09:27:42,456 [INFO]: exp_dir: ./experiments_BD/TecoGAN/TecoGAN_VimeoTecoGAN_4xSR_2GPU
2022-01-12 09:27:42,456 [INFO]: gpu_ids: 0
2022-01-12 09:27:42,456 [INFO]: is_train: False
2022-01-12 09:27:42,456 [INFO]: dist: False
2022-01-12 09:27:42,456 [INFO]: device: cuda
2022-01-12 09:27:42,456 [INFO]: rank: 0
2022-01-12 09:27:42,457 [INFO]: ========================================
2022-01-12 09:27:42,457 [INFO]: Testing model: TecoGAN_4x_BD_Vimeo_iter500K
2022-01-12 09:27:42,457 [INFO]: ========================================
2022-01-12 09:46:25,930 [INFO]: Generator: FRNet
FRNet(
  (upsample_func): BicubicUpsampler()
  (fnet): FNet(
    (encoder1): Sequential(
      (0): Conv2d(6, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
      (4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (encoder2): Sequential(
      (0): Conv2d(32, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
      (4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (encoder3): Sequential(
      (0): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
      (4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
    )
    (decoder1): Sequential(
      (0): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
    )
    (decoder2): Sequential(
      (0): Conv2d(256, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
    )
    (decoder3): Sequential(
      (0): Conv2d(128, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (3): LeakyReLU(negative_slope=0.2, inplace=True)
    )
    (flow): Sequential(
      (0): Conv2d(64, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): LeakyReLU(negative_slope=0.2, inplace=True)
      (2): Conv2d(32, 2, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    )
  )
  (srnet): SRNet(
    (conv_in): Sequential(
      (0): Conv2d(51, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
      (1): ReLU(inplace=True)
    )
    (resblocks): Sequential(
      (0): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (1): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (2): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (3): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (4): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (5): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (6): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (7): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (8): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
      (9): ResidualBlock(
        (conv): Sequential(
          (0): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
          (1): ReLU(inplace=True)
          (2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
        )
      )
    )
    (conv_up): Sequential(
      (0): ConvTranspose2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), output_padding=(1, 1))
      (1): ReLU(inplace=True)
      (2): ConvTranspose2d(64, 64, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), output_padding=(1, 1))
      (3): ReLU(inplace=True)
    )
    (conv_out): Conv2d(64, 3, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
    (upsample_func): BicubicUpsampler()
  )
)
2022-01-12 09:46:26,069 [INFO]: Load generator from: pretrained_models/TecoGAN_4x_BD_Vimeo_iter500K.pth
2022-01-12 09:46:26,070 [INFO]: Testing on Vid4 dataset
/home/haimou/桌面/project_all/TecoGAN-PyTorch/codes/utils/data_utils.py:87: RuntimeWarning: overflow encountered in multiply
  return np.uint8(np.clip(np.round(inputs * 255), 0, 255))
2022-01-12 09:48:04,296 [INFO]: ----------------------------------------
2022-01-12 09:48:04,296 [INFO]: Testing on ToS3 dataset
2022-01-12 09:48:42,102 [INFO]: ----------------------------------------
`

**Comment by Ruinmou** [comment]
The reason for 3090 and pytorch?

**Comment by jndeng** [comment]
According to the message, it could be the reason of 3090 and pytorch.
I also notice  that there's overflow encountered, which did not happen in my cases.

**Comment by Ruinmou** [comment]
thanks for your help
